autogluon/utils/tabular/ml/models/abstract/abstract_model.py

```diff
@@ -228,19 +233,45 @@ def compute_feature_importance(self, X, y, features_to_use=None):
         X.reset_index(drop=True, inplace=True)
         y.reset_index(drop=True, inplace=True)
 
-        X = self.preprocess(X)
+        if preprocess:
+            X = self.preprocess(X)
 
         if not features_to_use:
             features = X.columns.values
         else:
             features = features_to_use
-        feature_count = len(features)
 
-        model_score_base = self.score(X=X, y=y)
+        feature_importance_quick_dict = self.get_model_feature_importance()
+        # TODO: Also consider banning features with close to 0 importance
+        # TODO: Consider adding 'golden' features if the importance is high enough to avoid unnecessary computation when doing feature selection
+        banned_features = [feature for feature, importance in feature_importance_quick_dict.items() if importance == 0 and feature in features]
+        features = [feature for feature in features if feature not in banned_features]
+
+        permutation_importance_dict = self.compute_permutation_importance(X=X, y=y, features=features, preprocess=False)
+
+        feature_importances = pd.Series(permutation_importance_dict)
+        results_banned = pd.Series(data=[0 for _ in range(len(banned_features))], index=banned_features)
+        feature_importances = pd.concat([feature_importances, results_banned])
+        feature_importances = feature_importances.sort_values(ascending=False)
 
+        return feature_importances
+
+    # TODO: Consider repeating with different random seeds and averaging to increase confidence
+    # TODO: Optimize this
+    # TODO: Check memory usage to avoid OOM
+    # Compute feature importance via permutation importance
+    # Note: Expensive to compute
+    #  Time to compute is O(predict_time*num_features)
```

Comment:
Could also speed this up if need be by subsampling the data used to compute importances.  For instance, when `time_limits` are imposed and it is detected that this operation will take significant chunk of the remaining time otherwise.

Reply:
Yep, I already cap it at 10k rows, although thats fairly arbitrary.